modules/tensor_mechanics/src/dampers/ElementJacobianDamper.C

```diff
@@ -0,0 +1,136 @@
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementJacobianDamper.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "DisplacedProblem.h"
+#include "Assembly.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h" // _qrule
+
+template <>
+InputParameters
+validParams<ElementJacobianDamper>()
+{
+  InputParameters params = validParams<GeneralDamper>();
+  params.addClassDescription("Damper that limits the change in element Jacobians");
+  params.addParam<std::vector<NonlinearVariableName>>("displacements",
+                                                      "The nonlinear displacement variables");
+  params.addParam<Real>(
+      "max_increment",
+      0.1,
+      "The maximum permissible relative increment in the Jacobian per Newton iteration");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+ElementJacobianDamper::ElementJacobianDamper(const InputParameters & parameters)
+  : GeneralDamper(parameters),
+    _tid(parameters.get<THREAD_ID>("_tid")),
+    _assembly(_subproblem.assembly(_tid)),
+    _qrule(_assembly.qRule()),
+    _JxW(_assembly.JxW()),
+    _fe_problem(*parameters.getCheckedPointerParam<FEProblemBase *>("_fe_problem_base")),
+    _displaced_problem(_fe_problem.getDisplacedProblem()),
+    _max_jacobian_diff(parameters.get<Real>("max_increment"))
+{
+  if (_displaced_problem == NULL)
+    mooseError("ElementJacobianDamper: Must use displaced problem");
+
+  _mesh = &_displaced_problem->mesh();
+
+  const std::vector<NonlinearVariableName> & nl_vnames(
+      getParam<std::vector<NonlinearVariableName>>("displacements"));
+  _ndisp = nl_vnames.size();
+
+  for (unsigned int i = 0; i < _ndisp; ++i)
+  {
+    _disp_var.push_back(&_sys.getVariable(_tid, nl_vnames[i]));
+    _disp_incr.push_back(_disp_var.back()->increment());
+  }
+}
+
+void
+ElementJacobianDamper::initialSetup()
+{
+}
+
+Real
+ElementJacobianDamper::computeDamping(const NumericVector<Number> & /* solution */,
+                                      const NumericVector<Number> & update)
+{
+  // Maximum difference in the Jacobian for this Newton iteration
+  Real max_difference = 0.0;
+  MooseArray<Real> JxW_displaced;
+
+  // Vector for storing node coordinates before displacing them
+  std::vector<Point> node_copies;
```

Comment:
It would clarify things a little if we named this `point_copies` rather than `node_copies`.